harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -266,18 +277,20 @@ def lightning_scheduler_dict_to_det(lrs: dict) -> LRScheduler:
                 if lrs["interval"] == "epoch"
                 else LRScheduler.StepMode.STEP_EVERY_BATCH
             )
-            opt = getattr(lrs["scheduler"], "optimizer", None)
-            if opt is not None:
-                check.is_in(
-                    opt,
-                    self._pls.optimizers,
-                    "Must use an optimizer that is returned by wrap_optimizer()",
-                )
-            return LRScheduler(lrs["scheduler"], step_mode, frequency=lrs["frequency"])
 
-        optimizers = [self._pls.context.wrap_optimizer(opt) for opt in optimizers]
+            wrapped_opt = optimizers_dict[getattr(lrs["scheduler"], "optimizer", None)]
+            if wrapped_opt is None:
+                raise InvalidModelException("missing optimizer for lr_scheduler")
```

Comment:
As it is written, users who hit this message not know how to fix the issue or why they are encountering it.  Please make this message clearer as to what we are doing, why we are doing it, and how the user can work around the issue.

See `wrap_lr_scheduler()` for an example of what I mean:

```python
    raise det.errors.InvalidExperimentException(
        "detected that context.wrap_lr_scheduler() was called with an instance of "
        "torch.optim.lr_scheduer.ReduceLROnPlateau as the lr_scheduler.  This lr "
        "scheduler class does not have the usual step() parameters, and so it can "
        "only be used with step_mode=MANUAL_STEP.\n"
        "\n"
        "For example, if you wanted to step it on every validation step, you might "
        "wrap your lr_scheduler and pass it to a callback like this:\n"
        "\n"
        "class MyLRStepper(PyTorchCallback):\n"
        "    def __init__(self, wrapped_lr_scheduler):\n"
        "        self.wrapped_lr_scheduler = wrapped_lr_scheduler\n"
        "\n"
        "    def on_validation_end(self, metrics):\n"
        '        self.wrapped_lr_scheduler.step(metrics["validation_error"])\n'
    )
```

Reply:
Users will definitely break this.  I was helping a user who was passing a custom LRScheduler without a `step` method just a couple days ago.